_posts/2024-07-19-jberet-3_0_0_beta1-release.md

```diff
@@ -0,0 +1,40 @@
+---
+layout:     post
+title:      "The Release Note Of JBeret 3.0.0.Beta1"
+subtitle:   ""
+date:       2024-07-19
+author:     Wei Nan Li
+---
+
+Recently JBeret has a major version release::
+
+- [Release v3.0.0.Beta1 · jberet/jsr352](https://github.com/jberet/jsr352/releases/tag/3.0.0.Beta1)
+
+In this release it include these major changes:
+
+- [JBERET-445: JUnit5 Migration #467](https://github.com/jberet/jsr352/pull/467)
+- [JBERET-472: Extract database props from env-vars #473](https://github.com/jberet/jsr352/pull/473)
+
+And it added the tests against main branch of WildFly server, both preview and non-preview ones:
+
+- [JBERET-599: Setup the CI testings against WildFly main branch on #490](https://github.com/jberet/jsr352/pull/490)
+- [Add WildFly Preview CI Testing. #531](https://github.com/jberet/jsr352/pull/531)
+
+In addition, it removes some unnecessary API dependencies:
+
+- [Remove the dependency of jakarta.persistence-api #536](https://github.com/jberet/jsr352/pull/536)
+- [Remove the dependency of the validation-api #540](https://github.com/jberet/jsr352/pull/540)
+
+And it contains these Jakarta API related components upgrades:
+
+- [Bump jakarta.annotation:jakarta.annotation-api from 2.1.1 to 3.0.0 #506](https://github.com/jberet/jsr352/pull/501)
+- [Bump jakarta.enterprise:jakarta.enterprise.cdi-api from 4.0.1 to 4.1.0 #511](https://github.com/jberet/jsr352/pull/511)
+
+Because JBeret uses the basic usages of the `annotation-api` and the `cdi-api`, so generally speaking, the upgrade of the above two components doesn't introduce any major changes into the project. In addition, when the JBeret is used in an integrated environment, such as the `batch-jberet` module inside the WildFly, it will inherit the `annotation-api` and `cdi-api` versions used by WildFly itself. So this is a difference between JBeret SE and JBeret used inside an integrated environment like WildFly.
```

Comment:
I don't know that I would agree this doesn't introduce any major changes. 

The annotation-api less so, but I don't know the scope of the dependency. If it's `<scope>provide</scope>` and to use the user would need to add it. However, if the user relies on the dependency being transitive, they lose the `@ManagedBean` annotation.

For CDI, I'd argue that is a larger change. There are new features in the API which require an upgrade to Weld. IIRC there were some things removed too which if you implement parts of the API you must make changes to your code.

IoW just because JBeret doesn't use aspects of the API, doesn't mean users aren't using them transitively.

Reply:
Okay I see. Thanks for teaching James! I'll see how I make these info into the text.